aws/s3.tf

```diff
@@ -123,3 +123,94 @@ module "s3_bucket" {
 
   tags = local.tags
 }
+# aws S3 backup //////////////////////
+resource "aws_backup_vault" "datagrok_public_vault" {
+  name = var.s3_backup_vault_name
+}
+
+resource "aws_iam_role" "backup_role" {
+  name                = "s3-backup-role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Service = "backup.amazonaws.com"
+        }
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "s3_backup" {
+  name        = "backup-s3"
+  description = "policy for backup s3 datagrok-public bucket"
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:ListBucket",
+          "s3:GetBucketLocation",
+          "s3:GetObjectVersion",
+          "s3:GetObjectVersionAcl",
+          "s3:GetObject",
+          "s3:ListBucketMultipartUploads",
+          "s3:*",
+          "backup:CreateBackupPlan",
+          "backup:CreateBackupSelection",
+          "backup:CreateBackup",
+          "backup:StartBackupJob",
+          "backup:ListBackupPlans",
+          "backup:ListBackupSelections",
+          "backup:ListBackupVaults",
+          "cloudwatch:GetMetricData"
+        ]
+        Resource = [
+          module.s3_bucket.s3_bucket_arn,
+          "${module.s3_bucket.s3_bucket_arn}/*",
+          "*"
+        ]
+      }
+    ]
+  })
+}
+# Attach an IAM policy to the backup role that allows  performing AWS Backup jobs
+resource "aws_iam_role_policy_attachment" "backup_policy_attachment" {
+  policy_arn = aws_iam_policy.s3_backup.arn
+  role       = aws_iam_role.backup_role.name
+}
+resource "aws_iam_role_policy_attachment" "backup_service_role_policy" {
+  policy_arn = "arn:aws:iam::aws:policy/AWSBackupServiceRolePolicyForS3Backup"
+  role       = aws_iam_role.backup_role.name
+}
+resource "aws_backup_plan" "datagrok_public_s3_backup_plan" {
+  name = var.s3_backup_plan_name
```

Comment:
We already have a lot of variables; why would we need another one?
Use naming prefixes instead, like for logging bucket https://github.com/datagrok-ai/tf-module-datagrok-core/pull/26/files#diff-bfa811139a4cedd8c743561172e24c14f800b2fd054b137f984598b3d4195706R10